### ISSUE TITLE
Replacing PipelineResources in tutorial

### ISSUE THREAD

**Issue by psschwei** [tektoncd/pipeline#3705]
Currently, the Pipeline tutorial uses `git` and `image` PipelineResources for Task inputs/outputs [see here](https://github.com/tektoncd/pipeline/blob/master/docs/tutorial.md#specifying-task-inputs-and-outputs).  As there are beta-supported alternatives for both PipelineResources [for git](https://github.com/tektoncd/pipeline/blob/master/docs/migrating-v1alpha1-to-v1beta1.md#replacing-a-git-resource), [for image](https://github.com/tektoncd/pipeline/blob/master/docs/migrating-v1alpha1-to-v1beta1.md#replacing-an-image-resource)), it might be good to update the tutorial to use these alternatives.

**Comment by rudeigerc** [comment]
Hi, I'd like to make an update. However, the entire tutorial used `PipelineResource` as the resource input and output.
Should the update be imported as a complement of the origin tutorial or a brand new one?

**Comment by pritidesai** [comment]
We could add as complementing to the original tutorial that way folks can compare a pipeline with and without resources. @psschwei would that work?

**Comment by psschwei** [comment]
@pritidesai I like the idea of including both alternatives. However, for a new user looking to get started with Tekton (which I assume is the main audience for the tutorial), would it be better to keep things simple and only present a single way of doing things?

**Comment by pritidesai** [comment]
Sure @psschwei sounds good. 

How are we updating [this](https://github.com/tektoncd/pipeline/blob/master/docs/tutorial.md#specifying-task-inputs-and-outputs) section to replace pipeline resources? I am asking because this section refers to a task using pipeline resources to build an image. After replacing pipeline resources with the catalog tasks, it can no longer be represented as a single task instead it has to be converted to a pipeline to combine `git-clone` and `kaniko` or `buildah` catalog task 😞.

**Comment by vdemeester** [comment]
@pritidesai we would need to have a task that does all (the git-clone and the build). This means we wouldn't use a task from the catalog but it would work 👼🏼.

**Comment by psschwei** [comment]
> @pritidesai we would need to have a task that does all (the git-clone and the build). This means we wouldn't use a task from the catalog but it would work 👼🏼.

Does it make sense to add a clone&build task to the catalog?

**Comment by vdemeester** [comment]
> > @pritidesai we would need to have a task that does all (the git-clone and the build). This means we wouldn't use a task from the catalog but it would work 👼🏼.
> 
> Does it make sense to add a clone&build task to the catalog?

I am less sure about that *yet*. Let's see a bit where discussions on https://github.com/tektoncd/community/pull/316 and https://github.com/tektoncd/community/pull/318/files are going before 👼🏼 🙏🏼

**Comment by pritidesai** [comment]
> I am less sure about that _yet_. Let's see a bit where discussions on [tektoncd/community#316](https://github.com/tektoncd/community/pull/316) and https://github.com/tektoncd/community/pull/318/files are going before 👼🏼 🙏🏼

Yup, I agree with @vdemeester,  lets wait and see how the discussion unfolds on  tektoncd/community#316 and  tektoncd/community#318.

@psschwei you have hit this at the right time and I am hopeful we will find a solution to it.

**Comment by sbose78** [comment]
+1 

We need to start having folks get used to the alternatives of PipelineResources sooner.

**Comment by psschwei** [comment]
Adding link to [TEP-0044](https://github.com/tektoncd/community/blob/main/teps/0044-decouple-task-composition-from-scheduling.md) now that it's merged. It includes a comparison with [the TEP formerly known as 0046](https://github.com/tektoncd/community/pull/318), which was closed.

**Comment by jerop** [comment]
/lifecycle frozen

we still need to do this to help users migrate off pipelineresources

**Comment by jeremyjjbrown** [comment]
The staleness of such an important ticket (and easy?) makes me pause on using this project.

**Comment by rebelopsio** [comment]
I’m currently working through learning Tekton from a newcomer’s perspective including making use of workspaces. I’d love to help get the documentation updated around this topic.

**Comment by dibyom** [comment]
Thanks @rebelopsio I think @geriom also has some ideas in this space.

**Comment by geriom** [comment]
I created an [introductory tutorial last year](https://github.com/tektoncd/website/pull/298). I'll make my priority to update it and get it merged this week.

That wasn't merged because we struggled to decide where that doc fits, in the end we decided to step back and rethink the docs strategy, we (the documentation working group) came up with [a roadmap, new structure and new doc guidelines](https://github.com/tektoncd/website/pull/337) to make the overall documentation experience a lot better.

I'll open a bunch of issues with the proper context for every documentation item we have in mind. The idea is to have a good, friendly and educational documentation experience for the launch of pipelines v1.

**Comment by geriom** [comment]
Tracking this now in #4554 and #4555

**Comment by lbernick** [comment]
/assign geriom

Geri, sounds like you are working on this currently? feel free to remove your assignment if not.

**Comment by geriom** [comment]
@dibyom with #4789 merged I think we can close this issue. I'm don't have the permissions to do it.

**Comment by vdemeester** [comment]
/close
@geriom you should be able to close with the `/close` comment I think
